### ISSUE TITLE
[BUG] incorrect screeshots attached when running tests with fixtures

### ISSUE THREAD

**Issue by mk654321** [microsoft/playwright#26695]
### System info
- Playwright Version: [v1.37.0]
- Operating System: [linux]
- Browser: [All]
- Other info:

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Link to the GitHub repository with the repro**

[https://github.com/your_profile/playwright_issue_title]

or

**Config file**

```js
// playwright.config.ts
// @ts-check
const { defineConfig, devices } = require('@playwright/test');

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * @see https://playwright.dev/docs/test-configuration
 */
module.exports = defineConfig({
  testDir: './tests',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    // baseURL: 'http://127.0.0.1:3000',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
    /* screenshots on failure */
    screenshot: 'only-on-failure',

    //video: 'on'
  },

  /* Configure projects for major browsers */
  projects: [

    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },

    {
      name: 'firefox',
      use: { ...devices['Desktop Firefox'] },

    },

    {
      name: 'webkit',
      use: { ...devices['Desktop Safari'] },
    },

    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: { ...devices['Pixel 5'] },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: { ...devices['iPhone 12'] },
    // },

    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: { ..devices['Desktop Chrome'], channel: 'chrome' },
    // },
  ],

  /* Run your local dev server before starting the tests */
  // webServer: {
  //   command: 'npm run start',
  //   url: 'http://127.0.0.1:3000',
  //   reuseExistingServer: !process.env.CI,
  // },
});


```

**Fixture file (self-contained)**
```js
//in fixtures/simpleFixtureUsingBrowser.js
const base = require('@playwright/test');

exports.test = base.test.extend({

  fixtureA: async ({ browser }, use) => {

    // OPEN NEW BROWSER
    const page = await browser.newPage();
    // Open page
    await page.goto("https://playwright.dev/");
    let the_heading = page.getByRole('heading', { name: 'Playwright enables reliable end-to-end testing for modern web apps.' });
    await base.expect(the_heading).toBeVisible({timeout:15_000});
    let a="a";
    let b="b";
    await page.close();
    use({a,b});
  },
  fixtureB: [async ({fixtureA,page},use) => {
    await page.goto("https://github.com/");
    let search_box = page.getByRole('button', { name: 'Search or jump to...' });
    await base.expect(search_box).toBeVisible({timeout:15_000});
    use(page);
  },{ timeout: 60_000 }]

});
exports.expect = base.expect;
```
**Test file (self-contained)**

```js
// @ts-check
const { test, expect } = require(process.cwd()+'fixtures/simpleFixtureUsingBrowser.js');

test('test fixtureA', async ({fixtureA,page}) => {
  const {a,b} = fixtureA;
  await page.goto("https://bing.com/");
  let search_box = page.getByPlaceholder('Search the web');
  await expect(search_box).toBeVisible({timeout:10_000});
  await search_box.click();
  await search_box.fill(a+b);
  await expect(page.getByRole('link', { name: 'Docz' })).toBeVisible();
});

test('test fixtureB', async ({fixtureB,page}) => {

  await page.goto("https://www.microsoft.com/");
  await expect(page.getByRole('link', { name: 'CommunityTEST' })).toBeVisible();
});

```

**Steps**
- Run the test
- Test fill fail, this is ok
-  Inspect test report
-  Inspect attached screenshots

**Expected**

I would only expect screenshots that are due to test failures :
- test fixtureA should only contain screenshot of bing[.]com page
- test fixtureA should only contain screenshot of microsoft[.]com page

**Actual**

In addition to expected screenshots, screenshots from fixtures are being attached (notably from fixtureA)

**Comment by tejasv02** [comment]
We have seen this issue with a test result uploaded in currents.dev

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your contribution to our project. This issue has been closed due to its limited upvotes and recent activity, and insufficient feedback for us to effectively act upon. Our priority is to focus on bugs that reflect higher user engagement and have actionable feedback, to ensure our bug database stays manageable.

Should you feel this closure was in error, please create a new issue and reference this one. We're open to revisiting it given increased support or additional clarity. Your understanding and cooperation are greatly appreciated.
